mvcc/backend/backend.go

```diff
@@ -119,6 +120,13 @@ func NewDefaultBackend(path string) Backend {
 }
 
 func newBackend(bcfg BackendConfig) *backend {
+	// snapshotWarningTimeout is based on transferring 4 GB over a 1 gigabit/s connection
+	// with max tcp throughput of 949.285 Mbit/s limited by TCP overhead.
+	// every 4 gigabytes increases snapshotWarningTimeout by 35 seconds.
+	// e.g size(db) <= 4 GB takes < 35 seconds, size(db) <= 8 GB takes < 70 seconds.
+	factor := bcfg.MmapSize/(4*1024*1024*1024) + 1
```

Comment:
@xiang90 I am thinking about calculating the warning duration base on following.

1. When Snapshot(), get real db size via `tx.Size()`.
2. `snapshotWarningTimeout` equals to either min warning duration or expected transfer time calculated using `tx.Size()`.

note: `snapshotWarningTimeout` becomes a local var of `Snapshot()`